lib/java_buildpack/util/cache/cached_file.rb

```diff
@@ -33,7 +33,7 @@ class CachedFile
         # @param [String] uri a uri which uniquely identifies the file in the cache
         # @param [Boolean] mutable whether the cached file should be mutable
         def initialize(cache_root, uri, mutable)
-          key            = URI.escape(uri.sanitize_uri, ':/')
+          key            = URI.escape(uri.sanitize_uri, ':/&')
```

Comment:
No, no!  This was a perfect change that I'm happy to have.  Just most people never dive deeply enough into the code to know what they need to do.  More like this please.